### PR TITLE
Call for Artifacts

### PR DESCRIPTION
**Pull request by codeZeilen**
Adds the CfA to the page. This is still a draft, as it also removes the paragraph on artifacts from the CfP. We might want to keep the paragraph on artifacts in the CfP in case it is used for advertisement.

### CODE REVIEW

**Comment by codeZeilen** [comment]
Also: The timeline for artifact submissions is currently duplicated on the CfA and the timeline page. I would vote for removing the AE dates from the timeline page, as it draws attention away from the journal timeline, which should be the main timeline for interested authors.

File: cfa/index.markdown [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+The Artifact Evaluation Committee (AEC) takes the position of future researchers and then asks themselves how much this artifact would help them. Please see details of the outcomes of artifact evaluation (badges) for further guidance.
+
+### Submission Process
+
+All accepted articles (including accepted articles subject to minor revisions) are eligible to submit artifacts.
+
+Your submission should consist of three parts:
+
+1.  a Markdown-formatted README.md of your artifact,
```
Comment (krono) [inline cfa/index.markdown]: \`README.md\`

File: cfa/index.markdown [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+*   In some cases repeating the evaluation may take a long time. Reviewers may not reproduce full results in such cases.
+
+In some cases, the artifact may require specialized hardware (e.g., a CPU with a particular new feature, a specific class of GPU, or a cluster of GPUs). For such cases, authors should contact the Artifact Evaluation Co-Chairs ([Patrick Rein and Fabio Niephaus](mailto:patrick.rein@hpi.uni-potsdam.de,fabio.niephaus@hpi.uni-potsdam.de?subject=ProgrammingAE)) as soon as possible after their articles’ acceptance notification to work out how to make these possible to evaluate. One possible outcome is that the authors of the artifact requiring specialized hardware pay for a cloud instance with the hardware, which reviewers could access remotely.
+
+
+#### Reusable
+
+![reusable badge](https://www.acm.org/binaries/content/gallery/acm/publications/replication-badges/artifacts_evaluated_reusable_dl.jpg)
```
Comment (krono) [inline cfa/index.markdown]: We should review whether hotlinking these images or hosting them.

File: cfa/index.markdown [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+*   Missing dependencies, or poor documentation of dependencies. The single most effective way to avoid these sorts of issues ahead of time is to run the instructions independently on a fresh machine, VM, or Docker container.
+*   Comparing against existing tools on new benchmarks, but not including ways to reproduce the other tools’ executions.
+*   Not explaining how to interpret results. For instance, if an artifact ran successfully and produced the output that was the basis for the article, but without any way for reviewers to check the output for consistency with the article. Examples included generating a list of warnings without documenting which were true vs. false positives and generating large tables of numbers that are presented graphically in the article without providing a way to generate analogous visualizations.  
+
+
+### COI
```
Comment (krono) [inline cfa/index.markdown]: Let's write that out in the header

File: cfa/index.markdown [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+**Can my artifact be accepted if some of the article’s claims are not supported by the artifact, for example, if some benchmarks are omitted or the artifact does not include tools we experimentally compare against in the article?**
+
+In general yes (if good explanations are provided, as explained above), but if such claims are essential to the overall results of the article, the artifact will not receive the Functional or Reusable badge. As an extreme example, an artifact consisting of a working tool submitted with no benchmarks (e.g., if all benchmarks have sources that may not be redistributed) would not receive the functional or reusable badge. Note that the artifact can still receive the Available badge.
+
+**Why do we need to use Zenodo for the _Available_ badge? Why not GitHub?**
+
+Commercial repositories are unreliable in that there is no guarantee the evaluated artifact will remain available indefinitely. Users can delete public repositories or their accounts. And in addition to universities deleting departmental URLs over time, hosting companies also sometimes simply delete data.
```
Comment (krono) [inline cfa/index.markdown]: Maybe add that zenodo can automagically create a stable entity from a github release. This should aleviate the "why not gh" issue

File: cfa/index.markdown [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+All accepted articles (including accepted articles subject to minor revisions) are eligible to submit artifacts.
+
+Your submission should consist of three parts:
+
+1.  a Markdown-formatted README.md of your artifact,
+2.  a URL pointing to either
+    *   a single file containing the artifact (recommended) or
+    *   the address of a public source control repository,
+3.  a hash certifying the version of the artifact at submission time, either
+    *   an sha256 hash of the single file (use the sha256sum command-line tool to generate the hash) or
```
Comment (fniephaus) [inline cfa/index.markdown]:
```suggestion
    *   an sha256 hash of the single file (use the `sha256sum` command-line tool to generate the hash) or
```

File: cfa/index.markdown [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+Submissions to AE are voluntary. Articles evaluated positively in the AE process will receive a seal of approval to be included on the first page of the article.
+If authors agree to make their artifacts publicly available, we will publish and archive the artifact alongside their article via Zenodo as part of the [Zenodo ‹Programming› community](https://zenodo.org/communities/programming/).
+Articles will include references to their artifacts and artifacts references to the corresponding articles.
+ 
+Three badges are available, based on the [ACM artifact reviewing badges](https://www.acm.org/publications/policies/artifact-review-and-badging-current): functional, reusable, and available (details below).
```
Comment (fniephaus) [inline cfa/index.markdown]:
```suggestion
Three badges are available, based on the [ACM artifact reviewing badges](https://www.acm.org/publications/policies/artifact-review-and-badging-current): _Functional_, _Reusable_, and _Available_ (details below).
```

File: cfa/index.markdown [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+*   A list of claims from your article supported by the artifact, and how they are supported.
+*   A list of claims from your article _not_ supported by the artifact, and why not.
+
+Examples of unsupported claims: Performance claims cannot be reproduced in VM, authors are not allowed to redistribute specific benchmarks, etc.
+
+Artifact reviewers can use this documentation to base their reviews/evaluations on these specific claims. The reviewers will still also consider whether the provided evidence is adequate to support claims that the artifact works as expected.
+
+#### Step-by-Step Instructions
+The **Step-by-Step Instructions** explain how to reproduce any experiments or other activities that support the individual claims in your article. Write this for readers who have a deep interest in your work and are studying it to improve it or compare against it. If your artifact runs for more than a few minutes to exhibit the desired effect, point this out, note how long it is expected to run (roughly), and explain how to run it on smaller inputs. Reviewers may choose to run on smaller or larger inputs depending on available hardware.
```
Comment (fniephaus) [inline cfa/index.markdown]:
```suggestion
The "Step-by-Step Instructions" explain how to reproduce any experiments or other activities that support the individual claims in your article. Write this for readers who have a deep interest in your work and are studying it to improve it or compare against it. If your artifact runs for more than a few minutes to exhibit the desired effect, point this out, note how long it is expected to run (roughly), and explain how to run it on smaller inputs. Reviewers may choose to run on smaller or larger inputs depending on available hardware.
```

File: cfa/index.markdown [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+### Contents of the Artifact README.md
+Your README.md should consist of three parts:
+
+ * Getting started guide
+ * Overview of the claims
+ * For each claim: Step-by-Step instructions for how you propose to evaluate your artifact (with appropriate connections to the relevant sections of your article)
+
+#### Getting Started Guide
+
+Your **Getting Started Guide** should contain setup instructions (including, for example, a pointer to the VM player software, its version, passwords if needed, etc.) and basic testing of your artifact that you expect a reviewer to be able to complete in 30 minutes or less. Reviewers will follow all the steps in the guide during an initial kick-the-tires phase. The Getting Started Guide should be as simple as possible, and yet it should stress the key elements of your artifact. Anyone who has successfully followed the Getting Started Guide should have no technical difficulties with your artifact.
```
Comment (fniephaus) [inline cfa/index.markdown]:
```suggestion
Your "Getting Started Guide" should contain setup instructions (including, for example, a pointer to the VM player software, its version, passwords if needed, etc.) and basic testing of your artifact that you expect a reviewer to be able to complete in 30 minutes or less. Reviewers will follow all the steps in the guide during an initial kick-the-tires phase. The Getting Started Guide should be as simple as possible, and yet it should stress the key elements of your artifact. Anyone who has successfully followed the Getting Started Guide should have no technical difficulties with your artifact.
```

File: cfa/index.markdown [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+
+Where appropriate, include descriptions of and links to files (included in the archive) that represent expected outputs (e.g., the log files expected to be generated by your tool on the given inputs); if there are warnings that are safe to be ignored, explain which ones those are.
+
+### Packaging the Artifact
+
+When packaging your artifact, please keep in mind: a) how accessible you are making your artifact to other researchers and b) the fact that the AEC members will have a limited time in which to make an assessment of each artifact.
+
+We ask that you make the artifact available as a Docker image or a virtual machine (VM) image in OVF / OVA format containing the installed artifact and all of the necessary libraries installed. A VM provides an easily reproducible environment. It also helps the AEC have confidence that errors or other problems cannot cause harm to their machines.
+
+You should make your artifact available as a single archive file. Please use a widely available compressed archive format such as ZIP (.zip), tar and gzip (.tgz), or tar and bzip2 (.tbz2). Please use open formats for documents and preferably CSV or JSON for data.
```
Comment (fniephaus) [inline cfa/index.markdown]:
```suggestion
You should make your artifact available as a single archive file. Please use a widely available compressed archive format such as ZIP (`.zip`), tar and gzip (`.tgz`), or tar and bzip2 (`.tbz2`). Please use open formats for documents and preferably CSV or JSON for data.
```

File: cfa/index.markdown (hunk 2 above)
Comment (codeZeilen) [reply cfa/index.markdown]: I guess reviewing is more along the lines of: We should probably host them. Can I just add them to the assets folder? :)

Comment (krono) [reply cfa/index.markdown]: go ahead

**Comment by codeZeilen** [comment]
I will to merge this tonight, comments are welcome up until then (and also afterwards).

edit: I will ask someone to merge. I forgot that I can not merge myself.

**Comment by codeZeilen** [comment]
@krono Could you do me the favor of merging this? Thanks!

**Comment by fniephaus** [comment]
Looks great: https://programming-journal.org/artifact-evaluation/

**Comment by krono** [comment]
hmm, now that I see it, what about putting the faq next to the AE-call?
would look like the "call for papers" "pecial issues" on top of the CPF page

**Comment by codeZeilen** [comment]
Good idea. Will do and open a new PR.

**Pull request merged**